Makefile

```diff
@@ -1,23 +1,27 @@
 venv: venv/bin/activate
 
-venv/bin/activate: requirements.txt requirements_dev.txt
-	test -e venv/bin/activate || virtualenv venv
-	. venv/bin/activate; pip install -Ur requirements.txt; pip install -Ur requirements_dev.txt; python setup.py install
+REQ_DIR=pip-dep
+
+reqs: $(REQ_DIR)/requirements.txt $(REQ_DIR)/requirements_dev.txt $(REQ_DIR)/requirements_udacity.txt
+
+venv/bin/activate: reqs
+	test -e venv/bin/activate || python -m venv venv
+	. venv/bin/activate; pip install -Ur $(REQ_DIR)/requirements.txt; pip install -Ur $(REQ_DIR)/requirements_dev.txt; pip install -Ur $(REQ_DIR)/requirements_udacity.txt; python setup.py install
 	touch venv/bin/activate
 
 install_hooks: venv
 	venv/bin/pre-commit install
 
 notebook: venv
 	(. venv/bin/activate; \
-		python setup.py install; \
+		python setup.py install udacity; \
```

Comment:
I like that udacity is the default (That being said could we make this a parameter which gets passed to the make target since I think I may actually be the only person who uses this target and would like to be able to use other configurations)